src/coreclr/jit/lsra.cpp

```diff
@@ -737,6 +755,12 @@ LinearScan::LinearScan(Compiler* theCompiler)
         {
             availableRegs[i] = &availableDoubleRegs;
         }
+#ifdef TARGET_XARCH
+        else if (thisType == TYP_MASK)
+        {
+            availableRegs[i] = &availableMaskRegs;
+        }
```

Comment:
I think this is a general case like the other where integers are the most common, but they're at the bottom of the checked cases for the loop.

Presumably changing this to either a `switch` or to prioritize setting `availableIntRegs` as the first path would be beneficial so we don't have "one extra check" before hitting the path.

--------------------------

Actually, looking at this, this entire thing is just very unnecessarily expensive. We could track the classification in the `typelist.h` so we could do:
```cpp
#define DEF_TP(tn, nm, jitType, verType, sz, sze, asze, st, al, tf, ar) availableRegs[static_cast<int>(TYP_##tn)] = &ar;
#include "typelist.h"
#undef DEF_TP
    TYP_COUNT
```

This would involve no loops, no complex checks as to type kind/etc.

@kunalspathak, thoughts? Probably not that impactful overall, since its just part of the LSRA constructor; but still seems "meaningful" given we have around 25 `TYP_*` defines and this is doing 4-5 checks to initialize 17 of them and 3-4 checks to initialize 6 others.

Reply:
-- Not saying this needs to be handled in this PR, its something we can do as a follow up if its something we want to do.

Reply:
I have updated #83109 to take care of it.